### PR TITLE
Temp

### PR DESCRIPTION
**Pull request by AkashBoora**
Please review the code.

### CODE REVIEW

File: src/components/molecules/BookCard/BookCard.tsx [hunk 1]
```diff
@@ -91,36 +91,113 @@ export type Book = {
...
 export interface BookCardComponentProps {
   book: Book;
   typeOfCard?: string;
-  onFinishedClick: (arg: any) => void;
+  changeBookStatus: (arg: Book) => void;
   bookObject: Array<Book>;
+  isCategoryTab?: boolean;
+}
+
+const buttonPropTypes = {
```
Comment (Nisanth-Chilakamarri-ZeMoSo) [inline src/components/molecules/BookCard/BookCard.tsx]: Refactor this component, molecules should not have this. Try to have enums declared to store the defaults

File: src/components/organisam/Footer/Footer.tsx [hunk 2]
```diff
@@ -1,31 +1,32 @@
+import { Grid } from "@mui/material";
 import {  styled } from "@mui/styles";
 import { FooterNavData } from "../../../data/FooterNavData";
 import Theme from "../../../Themes/themes";
 import { LogoComponent } from "../../atoms/Logo/Logo";
 import { TypographyComponent } from "../../atoms/Typography/Typography";
-import { FooterNavListComponent } from "../FooterNavList/FooterNavList";
+import { FooterNavListComponent } from "../../molecules/FooterNavList/FooterNavList";
```
Comment (Nisanth-Chilakamarri-ZeMoSo) [inline src/components/organisam/Footer/Footer.tsx]: Why is this in organisms?

File: src/components/pages/BookDetailsPage/BookDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+export const BookDetailsPageComponent = (
+  props: BookDetailsPageComponentProps
+) => {
+  let url = window.location.href;
+  let id = Number(url.slice(34));
+  console.log(url);
+
+  console.log(id);
+  const [bookData, setBookData] = useState<Book>();
```
Comment (Nisanth-Chilakamarri-ZeMoSo) [inline src/components/pages/BookDetailsPage/BookDetails.tsx]: This is not a page component, this is an organism

File: src/components/pages/BookDetailsPage/BookDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  const [dataModifiedToggle,setDataModifiedToggle] = useState(false);
+
+  useEffect(() => {
+    axios
+      .get(`http://localhost:3000/dataBase/${id}`)
+      .then((res: any) => {
+        setBookData(res.data);
+      })
+      .catch((error: any) => console.log(error));
```
Comment (Nisanth-Chilakamarri-ZeMoSo) [inline src/components/pages/BookDetailsPage/BookDetails.tsx]: try to move the axios calls to separate folder API, call the appropriate API whenever needed

File: src/components/pages/CategoryResultsPage/CategoryResults.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+export const CategoryResultsComponent = () => {
+  let url = window.location.href;
+  let category = url.slice(30);
+  category = category.replaceAll("%20"," ");
+  console.log(url);
+  const [bookData, setBookData] = useState<Book[]>([]);
+  const [dataModifiedToggle, setDataModifiedToggle] = useState(false);
+
+  useEffect(() => {
+    axios
```
Comment (Nisanth-Chilakamarri-ZeMoSo) [inline src/components/pages/CategoryResultsPage/CategoryResults.tsx]: Same, this is not a page component. Move all Axios calls to different folder

File: src/components/pages/ExplorePage/ExplorePage.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  category = url.slice(30).replaceAll("%20", " ");
+
+  const [bookData, setBookData] = useState<Book[]>([]);
+  const [dataModifiedToggle, setDataModifiedToggle] = useState(false);
+  const [searchText, setSearchText] = useState("");
+  console.log(category);
+  useEffect(() => {
+    axios
+      .get("http://localhost:3000/dataBase")
+      .then((res: any) => {
```
Comment (Nisanth-Chilakamarri-ZeMoSo) [inline src/components/pages/ExplorePage/ExplorePage.tsx]: same

File: src/components/pages/MyLibraryPage/MyLibrary.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+import { RootTemplate } from "../../templates/Template";
+import axios from "axios";
+import { HomePageTabComponent } from "../../organisam/HomePageTabs/HomePageTabs";
+
+
+// const Entrepreneurship = (props: { book: data[]; setData: any }) => {
+export const MyLibraryPageComponent = () => {
+  const [bookData, setBookData] = useState<Book[]>([]);
+  const [dataModifiedToggle, setDataModifiedToggle] = useState(false);
+
```
Comment (Nisanth-Chilakamarri-ZeMoSo) [inline src/components/pages/MyLibraryPage/MyLibrary.tsx]: same

File: src/components/organisam/BookDetails/BookDetails.tsx [hunk 8]
```diff
@@ -1,43 +1,79 @@
...
+  let url = window.location.href;
+  let id = Number(url.slice(34));
+  console.log(url);
+
+
+  const [book, setBookData] = useState<Book>();
+  const [dataModifiedToggle,setDataModifiedToggle] = useState(false);
+
+  useEffect(() => {
+    console.log("passed id",id);
```
Comment (Nisanth-Chilakamarri-ZeMoSo) [inline src/components/organisam/BookDetails/BookDetails.tsx]: remove all console logs

File: src/components/organisam/Footer/Footer.tsx (hunk 2 above)
Comment (Nisanth-Chilakamarri-ZeMoSo) [reply src/components/organisam/Footer/Footer.tsx]: move this to molecules

**Review by Nisanth-Chilakamarri-ZeMoSo — CHANGES_REQUESTED** [review]

**Comment by Nisanth-Chilakamarri-ZeMoSo** [comment]
Resolve the code smells

**Comment by AkashBoora** [comment]
Nisanth Please check now!

**Review by Nisanth-Chilakamarri-ZeMoSo — APPROVED** [review]

**Pull request merged**